osbuild/solver/__init__.py

```diff
@@ -11,49 +19,56 @@ def dump(self):
         pass
 
     @abc.abstractmethod
-    def depsolve(self, arguments):
+    def depsolve(self, args: "DepsolveCmdArgs"):
         pass
 
     @abc.abstractmethod
-    def search(self, args):
+    def search(self, args: "SearchCmdArgs"):
         pass
 
 
 class SolverBase(Solver):
     # put any shared helpers in here
-    pass
-
-
-class SolverException(Exception):
-    pass
-
-
-class GPGKeyReadError(SolverException):
-    pass
-
-
-class TransactionError(SolverException):
-    pass
-
-
-class RepoError(SolverException):
-    pass
-
-
-class NoReposError(SolverException):
-    pass
-
-
-class MarkingError(SolverException):
-    pass
-
-
-class DepsolveError(SolverException):
-    pass
-
 
-class InvalidRequestError(SolverException):
-    pass
+    # Override this in the subclass
+    SOLVER_NAME = "unknown"
+
+    def __init_subclass__(cls, **kwargs):
+        super().__init_subclass__(**kwargs)
+        if cls.SOLVER_NAME == "unknown":
```

Comment:
I would, instead of repeating the same string twice refer to `== SolverBase.SOLVER_NAME` here.